documentation/source/regulator.rst

```diff
@@ -0,0 +1,97 @@
+and load resistor (instance name ``i_resistor``).
+
+.. literalinclude:: ../../examples/mixed_signal/hdl/regulator.sv
+   :caption: regulator.sv
+   :language: systemverilog
+
+
+.. literalinclude:: ../../examples/mixed_signal/hdl/regulator_block.vams
+   :caption: regulator_block.vams
+   :language: systemverilog
+
+
+The Testbench
+-------------
+ 
+The testbench consists of both an HDL part and a Python/cocotb part.
+
+The HDL part of the Testbench
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The testbench HDL part is written in SystemVerilog and instantiates the design described above as ``i_regulator``.
+It also contains a probe module for analog values as instance ``i_analog_probe`` —
+imagine this being a multimeter that you quickly connect to different nodes in the design, measuring either voltage or current.
+
+.. literalinclude:: ../../examples/mixed_signal/hdl/tb_rescap.sv
+   :caption: tb_rescap.sv
+   :language: systemverilog
+
+The probe module can capture voltage and current of a node specified by ``node_to_probe`` (a string in this module containing a hierarchical path).
+The capturing occurs whenever there is an edge on the logic signals ``probe_voltage_toggle`` or ``probe_current_toggle``.
+The captured values can be read on real-value signals ``voltage`` and ``current`` in this module.
+
+Here is the capture code for ``voltage`` with the "user-interface" highlighted:
+
+.. literalinclude:: ../../examples/mixed_signal/hdl/analog_probe_cadence.sv
+   :caption: analog_probe_cadence.sv
+   :language: systemverilog
+   :lines: 3-15
+   :emphasize-lines: 1-4
+   :dedent: 2
+
+.. note:: This analog probe module is currently only implemented for the Cadence Incisive and Xcelium simulators.
+            
+The cocotb part of the Testbench
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The cocotb part of the testbench provides functions to:
+
+* do the sampling of voltage and current of a given node (:meth:`~test_regulator.Regulator_TB.get_sample_data()`),
+* trim the regulator as close as possible to a target voltage within a given trim value range ( :meth:`~test_regulator.Regulator_TB.find_trim_val()`), and
+* plot the sampled data to a file (:meth:`~test_regulator.Regulator_TB.plot_data()`).
+
+FIXME: describe test_regulator_minimalist.py
+  
+The testcase first sets a ``vdd`` voltage and three different trim values, saving the resulting output voltage ``vout`` for each.
+It then runs an automatic trimming routine to find the best trim value for a given target voltage.
+
+.. todo:: Expand
+
+
+Reference Documentation
+-----------------------
+
+cocotb Testbench
+~~~~~~~~~~~~~~~~
+
+.. currentmodule:: test_regulator
+
+.. autoclass:: Regulator_TB
+
+    .. automethod:: get_sample_data(nodes, num=1, delay_ns=1)
+    .. automethod:: find_trim_val(probed_node, target_volt, trim_val_node, trim_val_signed=True, trim_val_min=None, trim_val_max=None)
+    .. automethod:: plot_data(datasets, graphfile="cocotb_plot.png")
```

Comment:
I think you can omit the arguments here - or use `:members: get_sample_data, find_trim_val, plot_data` instead

Reply:
In the past, Sphinx autodoc did not properly support coroutines.
Now, there is https://github.com/sphinx-doc/sphinx/pull/6295, released with Sphinx 2.1, but this still doesn't show method signatures for coroutines.